scripts/zkai_attention_kv_d16_native_width_gate.py

```
@@ -0,0 +1,529 @@
+    return json.dumps(value, sort_keys=True, separators=(",", ":"), ensure_ascii=False).encode("utf-8")
+
+
+def blake2b_commitment(value: Any, domain: str) -> str:
+    digest = hashlib.blake2b(digest_size=32)
+    digest.update(domain.encode("utf-8"))
+    digest.update(b"\0")
+    digest.update(canonical_json_bytes(value))
+    return f"blake2b-256:{digest.hexdigest()}"
+
+
+def validate_pair(
```

Ruff (0.15.12)

[warning] Too many branches (19 > 12)

(PLR0912)